controllers/cart_controller.py

```diff
@@ -0,0 +1,58 @@
+from config.database import engine
+from sqlalchemy import func, select
+from helpers.api_helper import APIHelper
+from models.product_table import product_table,review_table
+from models.cart_table import cart_table
+from dtos.cart_models import AddToCart, GetCartResponse,CartItem
+from helpers.api_helper import APIHelper
+
+
+
+# Add to Cart API
+def add_to_cart(request:AddToCart,token):
+    with engine.connect() as db:
+      
+        existing_cart = db.execute(select(cart_table).where(cart_table.c.product_id == request.product_id).where(cart_table.c.user_id == token.id)).fetchone()
+        if existing_cart:
+            db.execute(cart_table.update().where(cart_table.c.id == existing_cart.id).values(quantity=existing_cart.quantity+request.quantity))
+            db.commit()
+            return APIHelper.send_success_response(successMessageKey="Cart Updated Successfully")
+        new_cart = db.execute(cart_table.insert().values(
+            user_id=token.id, product_id=request.product_id, product_quantity=request.quantity))
+        db.commit()        
+        return APIHelper.send_success_response(successMessageKey="Product Added to Cart Successfully")
+
+
+def remove_from_cart(product_id,token):
+    with engine.connect() as db:
+        existing_cart = db.execute(select(cart_table).where(cart_table.c.product_id == product_id).where(cart_table.c.user_id == token.id)).fetchone()
+        if existing_cart:
+            db.execute(cart_table.delete().where(cart_table.c.id == existing_cart.id))
+            db.commit()
+            return APIHelper.send_success_response(successMessageKey="Product Removed from Cart Successfully")
+        return APIHelper.send_error_response(errorMessageKey="Product Not in Cart")
+    
+#Get Cart
+def get_cart(token):
+    with engine.connect() as db:
+        existing_cart = db.execute(select(cart_table).where(cart_table.c.user_id == token.id)).fetchall()
+        if existing_cart:
+            cart_items = []
+            for row in existing_cart:
+                product = db.execute(select(product_table).where(product_table.c.id == row.product_id)).fetchone()
+                if product:
+                    rating = round(db.execute(select(func.coalesce(func.avg(review_table.c.rating), 0)).where(review_table.c.product_id == row.product_id)).scalar(), 1)
+                    cart_item = CartItem(
+                        product_id=row.product_id,
+                        product_name=product.name,
+                        product_category=product.category,
+                        price=product.price,
+                        rating=rating,
+                        discount=product.discount,
+                        quantity=row.product_quantity,  # Add quantity from cart_table
+                        amount=product.price * row.product_quantity  # Add amount from cart_table
```

Comment:
How discount will be applied here in amount?

Reply:
i am doing that feat by decrement the price